Data/ByteString.hs

```diff
@@ -683,7 +683,7 @@ mapAccumR f acc (PS fp o len) = unsafeDupablePerformIO $ withForeignPtr fp $ \a
 -- | 'scanl' is similar to 'foldl', but returns a list of successive
 -- reduced values from the left. This function will fuse.
 --
--- > scanl f z [x1, x2, ...] == [z, z `f` x1, (z `f` x1) `f` x2, ...]
+-- > scanl f z [x1, x2, ...] == pack [z, z `f` x1, (z `f` x1) `f` x2, ...]
```

Comment:
If we're doing this we'd need to be consistent, so that'd be pack on both lists.